### ISSUE TITLE
"sync" behavior of az storage azcopy blob sync is not clear

### ISSUE THREAD

**Issue by stewartadam** [Azure/azure-cli#20346]
The use of "sync" is ambiguous and could be two-way or one-way.

Is it a one-way sync, local to remote? For example, if I run `az storage azcopy blob sync` with an empty local folder is the entire remote deleted?

In general, description should be worded in a way to illustrate what a user can expect in the following scenarios:
1. Files exist locally, but no on the remote
2. Files exist on the remote, but not locally
3. The same file exists locally and on the remote, but the remote is newer (by timestamp)
4. The same file exists locally and on the remote, but the local is newer (by timestamp)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 869e5e24-5afd-9d70-8d38-06efaf0dacee
* Version Independent ID: b939dad9-b602-bc64-5ab7-9f52568b117e
* Content: [az storage azcopy blob](https://docs.microsoft.com/en-us/cli/azure/storage/azcopy/blob?view=azure-cli-latest#az_storage_azcopy_blob_sync)
* Content Source: [latest/docs-ref-autogen/storage/azcopy/blob.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/master/latest/docs-ref-autogen/storage/azcopy/blob.yml)
* Service: **storage**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by yonzhan** [comment]
@evelyn-ys for awareness

**Comment by evelyn-ys** [comment]
@calvinhzy Can you take a look and update the help example?

**Comment by jonie001** [comment]
"Sync" is a one-way synchronization from local to container. Local files will not change.

The results of the four scenarios are as follows:

1. Files exist locally, but no on the remote: Files will be uploaded to the container, local files will not change.
2. Files exist on the remote, but not locally: Files will be deleted from the container, local files will not change.
3. The same file exists locally and on the remote, but the remote is newer (by timestamp): The file in the container are not 
updated, local file will also not be updated.
4. The same file exists locally and on the remote, but the local is newer (by timestamp): The file in the container will be updated to the newer local file, the file locally will not change.

**Comment by stewartadam** [comment]
Thanks for clarifying - it would be great to have this information in the docs and ideally, a succinct explanation or link to docs in `--help` as well.

**Comment by perlun** [comment]
Late to the party @jonie001, but why does step 3 and 4 only check by timestamp by default? Wouldn't it be much safer to perform checksum-based checks here instead?

(Asking because we ran into this in one of our CI jobs. Files in Azure Blob storage wouldn't be updated, probably because the remote timestamp was newer than the local one. When we added the `-- --compare-hash md5` flag, things started working much better. We already had `-- --put-md5` in place.)
